src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Sdk.Razor.CurrentVersion.targets

```diff
@@ -58,6 +58,11 @@ Copyright (c) .NET Foundation. All rights reserved.
       RazorCoreCompile
     </RazorCompileDependsOn>
 
+    <BuildDependsOn>
+      _InitialzePreserveCompilationContext;
+      $(BuildDependsOn)
+    </BuildDependsOn>
```

Comment:
Is there something more granular we can attach to?

Reply:
Might be able to wire this up to `PrepareForBuildDependsOn`.